src/microbe_stage/ColonyCompoundBag.cs

```diff
@@ -88,18 +88,20 @@ public float TakeCompound(Compound compound, float amount)
 
     public float AddCompound(Compound compound, float amount)
     {
-        var originalAmount = amount;
+        var totalAmountAdded = 0.0f;
```

Comment:
Why this change from a count down to a count up system? Do you still need the declining `amount` variable?

Reply:
Actually, rereading back, I realized the refactor in this file wasn't necessary *per se*, but the fact I was mislead in the first place (arguably because the `CompoundBag.cs` equivalent was wrong), was enough of a reason for me to propose a refactor.

Diminishing `amount` is still necessary to take more than you can. 

To be note that the refactor may do a few more operations, however. I'm not that attached to it, so I'm open to dropping the last commit changing this.

Reply:
If the code became clearer to you, that's an improvement.